ssl/s3_lib.c

```diff
@@ -4148,6 +4151,34 @@ const SSL_CIPHER *ssl3_choose_cipher(SSL *s, STACK_OF(SSL_CIPHER) *clnt,
     if (s->options & SSL_OP_CIPHER_SERVER_PREFERENCE || tls1_suiteb(s)) {
         prio = srvr;
         allow = clnt;
+#ifndef OPENSSL_NO_CHACHA
+        /* If ChaCha is at the top of the client preference list,
+           temporarily prioritize ChaCha ciphers in the servers list. */
+        if (s->options & SSL_OP_PRIORITIZE_CHACHA && sk_SSL_CIPHER_num(clnt) > 0) {
+            c = sk_SSL_CIPHER_value(clnt, 0);
+            if (c->algorithm_enc == SSL_CHACHA20POLY1305) {
+                /* Is client preferred, reprioritize... */
+                new_chacha = sk_SSL_CIPHER_new_null();
+                if (new_chacha != NULL) {
+                    /* Put ChaCha at the top */
+                    for (i = 0; i < sk_SSL_CIPHER_num(srvr); i++) {
+                        c = sk_SSL_CIPHER_value(srvr, i);
+                        if (c->algorithm_enc == SSL_CHACHA20POLY1305) {
```

Comment:
This `if` 's `{ }` seems not necessary...

Reply:
Yes, the coding standard indicates that braces are not needed for one-statement blocks.

Reply:
HOWEVER, if the code is more clear, feel free to put them. :)

Reply:
That thought did cross my mind...

Reply:
I'll hold off doing anything here, unless there are other required changes.